adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDDFunctions.scala

```diff
@@ -49,6 +51,65 @@ class NucleotideContigFragmentRDDFunctions(rdd: RDD[NucleotideContigFragment]) e
   }
 
   /**
+   * Save nucleotide contig fragments in FASTA format.
+   *
+   * @param fileName file name
+   */
+  def adamSaveAsFasta(fileName: String) = {
+
+    def isFragment(record: NucleotideContigFragment): Boolean = {
+      (Option(record.fragmentNumber), Option(record.numberOfFragmentsInContig)) match {
+        case (None, None)       => false
+        case (None, Some(m))    => false
+        case (Some(n), None)    => false
+        case (Some(n), Some(m)) => (m > 1)
+      }
+    }
+
+    def toFasta(record: NucleotideContigFragment): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(record.contig.contigName)
+      Option(record.description).foreach(n => sb.append(" ").append(n))
+      if (isFragment(record)) {
+        sb.append(" fragment %d of %d".format(record.fragmentNumber + 1, record.numberOfFragmentsInContig))
+      }
+      for (line <- Splitter.fixedLength(60).split(record.fragmentSequence)) {
+        sb.append("\n")
+        sb.append(line)
+      }
+      sb.toString
+    }
+
+    rdd.map(toFasta).coalesce(1).saveAsTextFile(fileName)
```

Comment:
I wouldn't coalesce here by default. Rather, I would allow the user to specify an optional argument with the number of partitions to coalesce down to.